src/reducers/deleteEmptyThought.ts

```diff
@@ -28,7 +28,14 @@ const deleteEmptyThought = (state: State): State => {
   }
   // archive an empty thought with hidden children
   else if (isEmpty && visibleChildren.length === 0) {
-    return archiveThought(state, {})
+    const archivedChild = allChildren[0]
+    return reducerFlow([
+      existingThoughtMove({ oldPath: [...simplePath, archivedChild], newPath: [...parentOf(cursor), archivedChild] }),
```

Comment:
You're using `simplePath` to create `oldPath`, but `parentOf(cursor)` to create `newPath`. They need to either both be based on `SimplePath` or `Path`. I think `Path` is probably the correct choice here so that the cursor is correctly restored within nested context views.